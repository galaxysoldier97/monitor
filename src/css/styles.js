import { typography } from '@material-ui/system';

const styles = {
  navigation: {
    paddingBottom: 15,
  },
  title: {
    fontSize: 24,
    fontWeight: typography.fontWeightLight,
    marginBottom: 20,
  },
  paper: {
    padding: 30,
  },
  clear: {
    clear: 'both',
  },
};

export default styles;
