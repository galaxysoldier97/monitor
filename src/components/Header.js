import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { AppBar, Button, Divider, Grid, IconButton, Menu, MenuItem, Toolbar, Typography } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import { makeStyles } from '@material-ui/styles';
import { Auth } from '../services/Auth';
import { drawerWidth } from '../config';
import { t } from 'mt-react-library/functions';
import { System } from '../data';
import Chip from '@material-ui/core/Chip';
import { getCompanyEnv, getCompanyEnvColor } from '../config/company';
import PropTypes from 'prop-types';

const useStyles = makeStyles(theme => ({
  appBar: {
    color: theme.palette.primary.contrastText,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    width: `100%`,
    paddingLeft: drawerWidth,
    transition: theme.transitions.create(['margin', 'width'], {
      easing: theme.transitions.easing.easeOut,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  appToolbar: {
    backgroundColor: theme.toolbar.bgColor,
    width: '100%',
  },
  menuButtonContainer: {
    flewGrow: 0,
    flewShrink: 0,
  },
  titleContainer: {
    flexGrow: 1,
  },
  userContainer: {
    flewGrow: 0,
    flewShrink: 0,
  },
  username: {
    textTransform: 'initial',
  },
}));

const Header = ({drawerOpen, handleDrawerOpen, handleDrawerClose}) => {
  const classes = useStyles();
  const [menuElement, setMenuElement] = useState(null);
  const username = Auth.getConnectedUser();
  return (
    <AppBar position="fixed" className={classNames(classes.appBar, {[classes.appBarShift]: drawerOpen})}>
      <Toolbar disableGutters={false} className={classes.appToolbar}>
        <Grid container alignItems="center">
          <Grid item className={classes.menuButtonContainer}>
            <IconButton color="inherit" onClick={drawerOpen ? handleDrawerClose : handleDrawerOpen}>
              <MenuIcon />
            </IconButton>
          </Grid>
          <Grid item className={classes.titleContainer}>
            <Grid container spacing={3} alignItems="center">
              <Grid item>
                <Typography variant="h5" noWrap color="inherit">Technical Repository Monitoring</Typography>
              </Grid>
              {!!getCompanyEnv() && <Grid item>
                <Chip label={getCompanyEnv()} style={{color: getCompanyEnvColor(), fontWeight: 600}} />
              </Grid>}
            </Grid>
          </Grid>
          <Grid item className={classes.userContainer}>
            <Button endIcon={<MoreVertIcon />} color="inherit" aria-haspopup="true" onClick={event => setMenuElement(event.currentTarget)}>
              <Typography className={classes.username}>{username}</Typography>
            </Button>
            <Menu anchorEl={menuElement} open={Boolean(menuElement)} onClose={() => setMenuElement(null)}>
              <MenuItem component={Link} to="/login" onClick={Auth.deauthenticateUser}>{t('header.signOut')}</MenuItem>
              {<Divider />}
              <MenuItem disabled display="block">
                <Typography align="center" display="block" color="secondary">
                  {System.version}
                </Typography>
              </MenuItem>
            </Menu>
          </Grid>
        </Grid>
      </Toolbar>
    </AppBar>
  );
};

Header.propTypes = {
  handleDrawerClose: PropTypes.func,
  handleDrawerOpen: PropTypes.func,
  drawerOpen: PropTypes.bool,
};

export default Header;
