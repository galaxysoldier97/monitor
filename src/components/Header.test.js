import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';
import { ThemeProvider } from '@material-ui/styles';
import { createMuiTheme } from '@material-ui/core';

test('Header snapshot', () => {
  const theme = createMuiTheme({toolbar:{bgColor:'#000000'}});
  const component = renderer.create(<ThemeProvider theme={theme}><Header /></ThemeProvider>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
