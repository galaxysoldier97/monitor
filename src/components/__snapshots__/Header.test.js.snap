// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular makeStyles-appToolbar-3 MuiToolbar-gutters"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root makeStyles-menuButtonContainer-4 MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root makeStyles-titleContainer-5 MuiGrid-item"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-align-items-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <h5
              className="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit MuiTypography-noWrap"
            >
              Technical Repository Monitoring
            </h5>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-userContainer-6 MuiGrid-item"
      >
        <button
          aria-haspopup="true"
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <p
              className="MuiTypography-root makeStyles-username-7 MuiTypography-body1"
            />
            <span
              className="MuiButton-endIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</header>
`;
